src/capellambse/helpers.py

```diff
@@ -578,6 +659,118 @@ def cb(el: etree._Element) -> None:
     return process_html_fragments(markup, cb)
 
 
+def _get_handler(
+    resources: cabc.Mapping[str, fh.FileHandler], name: str, /
+) -> fh.FileHandler | None:
+    return (
+        resources.get(name)
+        or resources.get(name.removesuffix(".team"))
+        or resources.get("\x00")
+    )
+
```

Comment:
I propose to add another helper function that reduces the duplication between both callback functions `cb` in embed and `unembed_images`:
```suggestion


def _get_handler_and_relpath(
    src: str, resources: cabc.Mapping[str, fh.FileHandler]
) -> tuple[fh.FileHandler, pathlib.PurePosixPath] | None:
    if not src or "://" in src:
        return None

    parts = pathlib.PurePosixPath(src).parts
    if len(parts) < 2:
        return None

    hdlname, *rest = parts
    relpath = pathlib.PurePosixPath(*rest)
    hdl = _get_handler(resources, hdlname)
    return (hdl, relpath) if hdl else None
```